gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiServiceImpl.java

```diff
@@ -553,11 +553,21 @@ public ApiEntity createWithApiDefinition(ExecutionContext executionContext, Upda
             repoApi.setVisibility(api.getVisibility() == null ? Visibility.PRIVATE : Visibility.valueOf(api.getVisibility().toString()));
 
             log.debug("Add default groups");
-            Set<String> defaultGroups = groupService
-                .findByEvent(executionContext.getEnvironmentId(), GroupEvent.API_CREATE)
-                .stream()
-                .map(GroupEntity::getId)
-                .collect(toSet());
+            Set<GroupEntity> defaultGroupEntities = groupService.findByEvent(executionContext.getEnvironmentId(), GroupEvent.API_CREATE);
+
+            Set<String> defaultGroups;
+            // Filter out groups with apiPrimaryOwner if primaryOwner is not null and has a value
+            if (primaryOwner != null && !StringUtils.isEmpty(primaryOwner.getId())) {
```

Comment:
As an API should always have a primary owner. Won't we always enter in the if and not in the else here ?